synapse/lib/trigger.py

```diff
@@ -25,78 +25,74 @@
     'prop:set',
 ))
 
-RecursionDepth = contextvars.ContextVar('RecursionDepth', default=0)
+@dataclasses.dataclass
+class Rule:
+    ver: int  # version: must be 1
+    cond: str  # condition from above list
+    useriden: str
+    storm: str  # storm query
+    viewiden: str # owning view
+    enabled: bool = True
+    form: Optional[str] = dataclasses.field(default=None) # form name
+    tag: Optional[str] = dataclasses.field(default=None) # tag name
+    prop: Optional[str] = dataclasses.field(default=None) # property name
+
+    def __post_init__(self):
+        if self.ver != 1:
```

Comment:
should we be version bumping this to account for the view iden?

Reply:
We only need to bump the version when we make a change that isn't backwards compatible. Adding a field is backwards compatible.